File: lib/domain/entities/product.dart
class Product {
  final int id;
  final String name;
  final String? description;
  final String? summaryDescription;
  final int? discount;
  final int brandId;
  final int categoryId;
  final double price;
  final String? avatar;
  Product({
    required this.id,
    required this.name,
    this.description,
    this.summaryDescription,
    this.discount,
    required this.brandId,
    required this.categoryId,
    required this.price,
    this.avatar,
  });
}

File: server/src/app/models/Product.js
const { DataTypes } = require('sequelize');
const sequelize = require('../../config/db');
const Brand = require('./Brand');
const Category = require('./Category');

const Product = sequelize.define('Product', {
    ProductId: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
    },
    ProductName: {
        type: DataTypes.STRING,
        allowNull: false,
    },
    Description: {
        type: DataTypes.TEXT,
        allowNull: true,
    },
    SummaryDescription: {
        type: DataTypes.TEXT,
        allowNull: true,
    },
    Discount: {
        type: DataTypes.INTEGER,
        allowNull: true,
    },
    IdBrand: {
        type: DataTypes.INTEGER,
        allowNull: false,
    },
    CategoryId: {
        type: DataTypes.INTEGER,
        allowNull: false,
    },
    Price: {
        type: DataTypes.FLOAT,
        allowNull: false,
    },
    Avatar: {
        type: DataTypes.STRING,
        allowNull: true,
    },
}, {
    tableName: 'product',
    timestamps: false,
    freezeTableName: true,
});

// Thiết lập mối quan hệ
Product.belongsTo(Brand, { foreignKey: 'IdBrand' });
Brand.hasMany(Product, { foreignKey: 'IdBrand' });

Product.belongsTo(Category, { foreignKey: 'CategoryId' });
Category.hasMany(Product, { foreignKey: 'CategoryId' });

module.exports = Product;
